### PR TITLE
Add Nim version 1.6.16 and 2.0.0

### PR DESCRIPTION
**Pull request by adokitkat**
Related https://github.com/compiler-explorer/compiler-explorer/pull/5702

### CODE REVIEW

**Review by jeremy-rifkin — APPROVED** [review]
Lgtm

File: bin/yaml/nim.yaml [hunk 1]
```diff
@@ -16,6 +16,8 @@ compilers:
       - 1.4.6
       - 1.4.8
       - 1.6.0
+      - 1.6.16
```
Comment (richardburce19) [inline bin/yaml/nim.yaml:19]: >

**Review by richardburce19 — COMMENTED** [review]
#1190

File: bin/yaml/nim.yaml [hunk 2]
```diff
@@ -16,6 +16,8 @@ compilers:
...
+      - 2.0.0
```
Comment (richardburce19) [inline bin/yaml/nim.yaml:20]: /yaml/nim.yaml

**Pull request merged**